### DISCUSSION TITLE
Command in TemplatedControl - Call from invalid thread error

### DISCUSSION THREAD

**Discussion by Basnederveen** [AvaloniaUI/Avalonia#12268] [Q&A] [answered]
I'm trying to create a TemplatedControl with a command in the code-behind. I want to accomplish a generic Folder Browse control with a textbox which displays the Path property and a button which activates the FolderPicker and puts the result in the Path property. 

I create the command using ReactiveCommand.Create() in the constructor of the control. (also tried CreateFromTask, same result)

When I execute the command through the UI, an InvalidOperationException (Call from invalid thread) is thrown even when I wrap the execution method in Dispatcher.UIThread.InvokeAsync or Post. 

Is it at all possible to place a command in the templatedcontrol to get a 'default' behaviour, without binding anything?

Test repo:
https://github.com/Basnederveen/FolderBrowseControlThreadError/tree/master

**Comment by timunie** [comment, marked as answer]
I filed a PR. Basically, don't create properties if you don't need to bind to it. Better to use event listener here in that specific use case.

**Reply by Basnederveen** [reply]
Thank you!
